api/src/public/api.json

```diff
@@ -0,0 +1,430 @@
+                  "description": "The ID of the next page in the course, not applicable for the last page in the course",
+                  "$ref": "#/components/schemas/Id"
+                },
+                "previous": {
+                  "description": "The ID of the previous page in the course, not applicable for the first page in the course",
+                  "$ref": "#/components/schemas/Id"
+                }
+              },
+              "readOnly": true,
+              "required": [
+                "id",
+                "type"
+              ]
+            },
+            "minItems": 1
+          }
+        },
+        "readOnly": true,
+          "required": [
+            "id",
+            "name",
+            "firstPage",
+            "pages"
+          ]
+        
+      },
+      "Page": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "description": "The unique ID for the page",
+            "$ref": "#/components/schemas/Id"
+          },
+          "title": {
+            "description": "The title for the page",
+            "type": "string"
+          },
+          "type": {
+            "description": "The type of the page",
+            "type": "string",
+            "enum": [
+              "learn",
+              "practice"
+            ]
+          },
+          "sections": {
+            "description": "The sections of the page, only applicable for learn type pages",
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/Section"
+            }
+          },
+          "problem": {
+            "description": "The practice problem for the page, only applicable for practice type pages ",
+            "type": "object",
+            "properties": {
+              "type": {
+                "description": "The type of the practice problem",
+                "type": "string",
+                "enum": [
+                  "code",
+                  "text",
+                  "multichoice"
+                ]
+              },
+              "question": {
+                "description": "The problem question prompt to display",
+                "type": "string"
+              },
+              "data": {
+                "description": "The data used to display and submit a solution to the problem, dependent on problem type",
+                "type": "object"
+              }
+            },
+            "required": [
+              "type",
+              "question",
+              "data"
+            ],
+            "readOnly": true
+          }
+        },
+        "required": [
+          "id",
+          "type"
+        ],
+        "readOnly": true
+      },
+      "Submission": {
+        "type": "object",
+        "properties": {
+          "type": {
+            "description": "The type of the submission",
+            "type": "string",
+            "enum": [
+              "code",
+              "text",
+              "eventStream"
```

Comment:
In the case of a multichoice problem(which appears in Figma), do we need another submission type for that type of problem? I'm assuming the 'eventStream' is for future implementation of dynamic algorithm visualization upon submitting blockly program

Reply:
yep, good thinking, I suppose this is still pretty unclear. I had imagined that  each problem would have a submission type, so yeah i think we could start linking those up and also start defining what their structure should be.